dashboard/src/modules/components/ProfileComponent/NewKeyModal.jsx

```diff
@@ -0,0 +1,63 @@
+import "./index.less";
+
+import {
+  Button,
+  Form,
+  FormGroup,
+  Modal,
+  ModalVariant,
+  TextInput,
+} from "@patternfly/react-core";
+import {
+  sendNewKeyRequest,
+  setNewKeyLabel,
+} from "actions/keyManagementActions";
+import { useDispatch, useSelector } from "react-redux";
+
+import React from "react";
+
+const NewKeyModal = (props) => {
+  const dispatch = useDispatch();
+  const { isModalOpen, newKeyLabel } = useSelector(
+    (state) => state.keyManagement
+  );
+
+  const createNewKey = () => {
+    dispatch(sendNewKeyRequest(newKeyLabel));
+  };
+  return (
+    <Modal
+      variant={ModalVariant.small}
+      title="New API Key"
+      isOpen={isModalOpen}
+      showClose={false}
+      actions={[
+        <Button
+          key="create"
+          variant="primary"
+          form="modal-with-form-form"
+          onClick={createNewKey}
```

Comment:
Rather than defining `createNewKey()`, I would use an in-line anonymous function, like you did at line 55:
```javascript
          onClick={() => dispatch(sendNewKeyRequest(newKeyLabel))}

```

Reply:
Got it!